process/economics/apiEconomicsData.go

```diff
@@ -0,0 +1,36 @@
+package economics
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+type apiEconomicsData struct {
+	*economicsData
+}
+
+// NewAPIEconomicsData will create a wrapped object over the provided economics data
+func NewAPIEconomicsData(data *economicsData) (*apiEconomicsData, error) {
+	if data == nil {
+		return nil, process.ErrNilEconomicsData
+	}
+
+	return &apiEconomicsData{
+		economicsData: data,
+	}, nil
+}
+
+// CheckValidityTxValues checks if the provided transaction is economically correct. It overloads the original method
```

Comment:
`if the provided transaction is economically correct` => `if the provided transaction respects the gas limit restrictions`

Reply:
actually it also checks other stuff other than the gas limit as it calls the overloaded method